fmriprep/workflows/bold/base.py

```diff
@@ -346,7 +346,12 @@ def init_func_preproc_wf(bold_file, ignore, freesurfer,
         name='inputnode')
     inputnode.inputs.bold_file = bold_file
     if sbref_file is not None:
-        inputnode.inputs.sbref_file = sbref_file
+        from niworkflows.interfaces.images import ValidateImage
+        val_sbref = pe.Node(ValidateImage(), name='val_sbref')
+        val_sbref.inputs.in_file = sbref_file
+        workflow.connect([
+            (val_sbref, inputnode, [('out_file', 'sbref_file')]),
+        ])
```

Comment:
Weren't we already doing a post-estimation validation to catch the sbref case? If we're going to do a pre- for both sbref and bold, then we should drop the post.

Reply:
Also, just from a conceptual workflows-as-functiony-things perspective, I'm not a huge fan of setting a new node to input to `inputnode`. I'm not sure if it's more maintainable to insert `val_sbref` between `inputnode` and `bold_reference_wf` or not.

Reply:
I'm not a fan of that either, but I couldn't see other alternatives. Only passing the sbref_file as an argument to the workflow, and I very much disliked the idea. I was tempted to fix that for the bold_file itself but understood that fix should go on a different PR.

Reply:
What about removing `sbref_file` from `inputnode`, and just connecting `val_sbref` to `bold_reference_wf`? Since we only populate the inputnode from the argument anyway, there's no real advantage to it being there.

Reply:
I have no issues with that.